docs/design.md

```diff
@@ -0,0 +1,79 @@
+Given that enormous thought went into rules_docker, one can't simply understand how it behaves vs how it supposed to behave. Basically, it is just a black box.
+
+
+## Why it is a good time to move away from docker v2 images
+we can all agree that docker revolutionized the container technology and changed the way we deploy. but this does not mean that we have to stick with it. there are other 
+container runtimes as well, such as cri-o, containerd (extracted from dockerd), youki which implements oci-runtime spec (which in turn targets oci-image-spec) rather than docker v2 image spec. 
+
+docker v2 spec which preceded docker v1 is the widely used container format out there. One downside is that they still inherits some properties of old dockerd such as HEALTHCHECK, RESOURCE LIMITS, but these properties not easily portable.
+
+open container initiative has started https://github.com/opencontainers/image-spec standardize how container image should be built. docker v2 images almost identical to oci counterparts except that they use different media types and contains some docker specific configuration properties. but nevertheless we want to stick with OCI format to avoid maintaining things that are specific to docker. for instance: https://github.com/bazelbuild/rules_docker/pull/1742
+
+the OCI image-spec format has been widely adopted by container runtimes and supported by all major distribution registries. so there is not point in supporting something that is preceded by OCI image-spec. Also, this makes our jobs a lot easier since we know what features to built and conform.
+
+Blog posts and docs to read:
+https://docs.docker.com/registry/spec/manifest-v2-2/
+https://kubernetes.io/blog/2020/12/02/dont-panic-kubernetes-and-docker/
+
+## What we do different than rules_docker
+From users perspective, not much if we are being honest. In fact users gonna be pissed off because we won't have 100% percent feature parity with rules_docker but this is for the best. we will drop anything that is docker specific. we won't have rules such as `container_run_and_extract` due to lack of a running runtime
+deamon available to us.
+
+Instead of maintaining programs that builds the images ourselves, we will always rely on existing tools such as `crane` or `umoci`, `skopeo` whenever possible. these tools are great candidate for usage with bazel since they directly work with in-disk CAS store which gives us great ability to build image without needing a running runtime deamon or remote registries. 
+
+These tools will give us the ability to be more inline with the container ecosystem because they are maintained by gifted people who also drive the image spec and container industry. On the other hand, with the time spared from not having to maintain these tools by ourselves, we could give back to these great tools.
```

Comment:
The maintainers of those projects are definitely gifted people. The maintainers of rules_docker are also gifted people. Again, be more precise:

```
they are well-maintained and have become the industry standard (because they are backed by OCI?)
```